In [40]:
# TODO: given loan amount, interest rate, and payment amount, enter them below and the amount payed in interst will be calculated
loan_amount = 500000
interest_rate = .06495
payment = 4354



payment_number = 1
total_interest_payed = 0
principal = loan_amount

while principal > 0:
    interest_payed = principal * (interest_rate/12)
    total_interest_payed += interest_payed
    principal_payed = payment - interest_payed
    # the commented out section below prints the balance and interest payed after every month
    # if (payment_number % 12 == 0):  
        # print("Payment %d: balance: $%.2f, $%.2f in interest payed, $%.2f in principal payed" % (payment_number, principal, interest_payed, principal_payed))
    principal -= principal_payed
    payment_number += 1

print("number of years:    %d" % (payment_number/12))
print("total loan amount:    %8.2f" % loan_amount)
print("total interest payed: %8.2f" % total_interest_payed)
print("Percent of loan payed in intereset: %.2f%%" % (total_interest_payed / loan_amount * 100))
print("number of years to pay loan if there was no interest:", (loan_amount / payment / 12))



number of years:    15
total loan amount:    500000.00
total interest payed: 283769.60
Percent of loan payed in intereset: 56.75%
number of years to pay loan if there was no interest: 9.569744296432399


## Some examples:


### brand new  BMW 2301 Coupe with a 60 month term
Costs:                  $38,200
Down Payment:           $2,500
Interest Rate:          5.99% 
Monthly Payment:        $690.02

Total Interest Payed:   $5,700.85
Percent of loan:        15.97%

### brand new  BMW 2301 Coupe with a 30 month term
Costs:                  $38,200
Down Payment:           $2,500
Interest Rate:          5.99% 
Monthly Payment:        $1,284.29

Total Interest Payed:   $2,828.58
Percent of loan:        7.92%


### $500k loan on a 30 year mortgage
Costs:                  $500,000
Down Payment:           $0
Interest Rate:          7.489% 
Monthly Payment:        $3492

Total Interest Payed:   $757,535.57
Percent of loan:        151.51%

### $500k loan on a 15 year mortgage
Costs:                  $500,000
Down Payment:           $0
Interest Rate:          6.495% 
Monthly Payment:        $4354

Total Interest Payed:   $283,769.60
Percent of loan:        56.75%


# Takeaways

- Every marginal decrease in interest rate or number of installments significantly decreases the total amount of interest paid over the course of the loan.
- Essentailly, the only way for a property to 'pay for itself' would be for it to increase in value the same amount as the interst paid on the loan by the time it is paid off.
